chaoscenter/authentication/api/main.go

```diff
@@ -0,0 +1,180 @@
+	"flag"
+	"fmt"
+	grpcHandler "litmus/litmus-portal/authentication/api/handlers/grpc"
+	grpcPresenter "litmus/litmus-portal/authentication/api/presenter/protos"
+	"litmus/litmus-portal/authentication/api/routes"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/misc"
+	"litmus/litmus-portal/authentication/pkg/project"
+	"litmus/litmus-portal/authentication/pkg/services"
+	"litmus/litmus-portal/authentication/pkg/user"
+	"litmus/litmus-portal/authentication/pkg/utils"
+	"net"
+	"runtime"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/kelseyhightower/envconfig"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type Config struct {
+	JwtSecret     string `required:"true" split_words:"true"`
+	AdminUsername string `required:"true" split_words:"true"`
+	AdminPassword string `required:"true" split_words:"true"`
+	DbServer      string `required:"true" split_words:"true"`
+	DbUser        string `required:"true" split_words:"true"`
+	DbPassword    string `required:"true" split_words:"true"`
+}
+
+func init() {
+	printVersion()
+
+	var c Config
+
+	err := envconfig.Process("", &c)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	// send logs to stderr so we can use 'kubectl logs'
+	_ = flag.Set("logtostderr", "true")
+	_ = flag.Set("v", "3")
+
+	flag.Parse()
+
+	client, err := utils.MongoConnection()
+	if err != nil {
+		log.Fatal("database connection error $s", err)
+	}
+
+	db := client.Database(utils.DBName)
+
+	// Creating User Collection
+	err = utils.CreateCollection(utils.UserCollection, db)
+	if err != nil {
+		log.Fatalf("failed to create collection  %s", err)
+	}
+
+	err = utils.CreateIndex(utils.UserCollection, utils.UsernameField, db)
+	if err != nil {
+		log.Fatalf("failed to create index  %s", err)
+	}
+
+	// Creating Project Collection
+	err = utils.CreateCollection(utils.ProjectCollection, db)
+	if err != nil {
+		log.Fatalf("failed to create collection  %s", err)
+	}
+
+	userCollection := db.Collection(utils.UserCollection)
+	userRepo := user.NewRepo(userCollection)
+
+	projectCollection := db.Collection(utils.ProjectCollection)
+	projectRepo := project.NewRepo(projectCollection)
+
+	miscRepo := misc.NewRepo(db, client)
+
+	applicationService := services.NewService(userRepo, projectRepo, miscRepo, db)
+
+	validatedAdminSetup(applicationService)
+
+	go runGrpcServer(applicationService)
+	runRestServer(applicationService)
+}
+
+func validatedAdminSetup(service services.ApplicationService) {
+	configs := map[string]string{"ADMIN_PASSWORD": utils.AdminPassword, "ADMIN_USERNAME": utils.AdminName, "DB_USER": utils.DBUser, "DB_SERVER": utils.DBUrl, "DB_NAME": utils.DBName, "DB_PASSWORD": utils.DBPassword, "JWT_SECRET": utils.JwtSecret}
+	for configName, configValue := range configs {
+		if configValue == "" {
+			log.Fatalf("Config %s has not been set", configName)
+		}
+	}
```

Comment:
This shouldn't be required if we already have all these fields as `required` in envconfig in line 29. envConfig itself will give error if they are not provided being required.

Reply:
Yes good catch! looks like duplicate check.